ANSI code being printed to terminal when using alacritty

<!-- Thank you for taking the time to fill out a bug report. -->

In edit mode `taskwarrior-tui` crashes after exiting the editor, whether there are changes or not.

<!-- A clear and concise description of what the bug is with screenshots if available. -->

**Steps to reproduce the behaviour:**

- Create empty `taskrc` and database:
```
$ mkdir tasktmp
$ touch tmprc
```

- Create a test task:
```
$ TASKRC=tmprc TASKDATA=tasktmp task add test
```

- Start `taskwarrior-tui`:
```
$ TASKRC=tmprc TASKDATA=tasktmp taskwarrior-tui
```
- Select the _test_ task and open the editor with `e`

- Quit the editor

_Output:_
```
Launching '/usr/bin/nvim "task.131964a7.task"' now...
Editing complete.
No edits were detected.
TASKRC override: /home/bexie/tmprc
TASKDATA override: /home/bexie/tasktmp
[taskwarrior-tui error]: Failed to create a Task from JSON. Please report as a github issue on https://github.com/kdheepak/taskwarrior-tui
```

_Note:_
Setting `RUST_BACKTRACE=1` does not produce any additional output.

**Environment:**

- Operating System: Arch Linux
- Installation: `taskwarrior-tui` package from the AUR
- taskwarrior-tui version: 0.9.3

- task version: 2.5.1

So, I tried adding the `dbg!` instructions you asked for, but when I try to compile I get errors like this

```
error[E0382]: borrow of moved value: `task`
   --> src/app.rs:877:21
    |
856 |         let mut task = Command::new("task");
    |             -------- move occurs because `task` has type `std::process::Command`, which does not implement the `Copy` trait
...
862 |         dbg!(task);
    |         ----------- value moved here
...
877 |                     task.arg(&s);
    |                     ^^^^ value borrowed here after move
```

Unfortunately, I am not familiar with rust and I have no clue how to work around this...

--- a/src/app.rs
+++ b/src/app.rs
@@ -1255,6 +1255,8 @@ impl TTApp {
                                 String::from_utf8_lossy(&output.stderr),
                             ))
                         } else {
+                            String::from_utf8_lossy(&output.stdout);
+                            String::from_utf8_lossy(&output.stderr);
                             Ok(())
                         }
                     }
@@ -1404,7 +1406,6 @@ impl TTApp {
         terminal: &mut Terminal<CrosstermBackend<io::Stdout>>,
         events: &Events,
     ) -> Result<(), Box<dyn Error>> {
-        events.pause_ticker();
         match self.mode {
             AppMode::TaskReport => match input {
                 Key::Ctrl('c') | Key::Char('q') => self.should_quit = true,
@@ -1640,7 +1641,6 @@ impl TTApp {
                 _ => {}
             },
         }
-        events.resume_ticker();
         Ok(())
     }
 }

--- a/src/main.rs
+++ b/src/main.rs
@@ -71,7 +71,7 @@ fn tui_main(_config: &str) -> Result<(), Box<dyn Error>> {
 
     // Setup event handlers
     let events = Events::with_config(EventConfig {
-        tick_rate: Duration::from_millis(1000),
+        tick_rate: Duration::from_millis(250),
     });
 
     let maybeapp = TTApp::new();

--- a/src/util.rs
+++ b/src/util.rs
@@ -9,7 +9,7 @@ use tui::{backend::CrosstermBackend, Terminal};
 use std::io::{self, Write};
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Arc;
-use std::{sync::mpsc, thread, time::Duration};
+use std::{sync::mpsc, thread, time::Duration, time::Instant};
 
 #[derive(Debug, Clone, Copy)]
 pub enum Key {
@@ -63,7 +63,6 @@ pub struct Events {
     pub rx: mpsc::Receiver<Event<Key>>,
     pub tx: mpsc::Sender<Event<Key>>,
     pub pause_stdin: Arc<AtomicBool>,
-    pub pause_ticker: Arc<AtomicBool>,
 }
 
 impl Events {
@@ -72,84 +71,67 @@ impl Events {
         use crossterm::event::{KeyCode::*, KeyModifiers};
         let (tx, rx) = mpsc::channel();
         let pause_stdin = Arc::new(AtomicBool::new(false));
-        let pause_ticker = Arc::new(AtomicBool::new(false));
+        let tick_rate = config.tick_rate;
         let _input_handle = {
             let tx = tx.clone();
             let pause_stdin = pause_stdin.clone();
             thread::spawn(move || {
+                let mut last_tick = Instant::now();
                 loop {
+                    let timeout = tick_rate
+                        .checked_sub(last_tick.elapsed())
+                        .unwrap_or_else(|| Duration::from_secs(0));
+
                     if pause_stdin.load(Ordering::SeqCst) {
                         thread::sleep(Duration::from_millis(250));
                         continue;
                     }
+
                     // poll for tick rate duration, if no event, sent tick event.
-                    if let event::Event::Key(key) = event::read().unwrap() {
-                        let key = match key.code {
-                            Backspace => Key::Backspace,
-                            Enter => Key::Char('\n'),
-                            Left => Key::Left,
-                            Right => Key::Right,
-                            Up => Key::Up,
-                            Down => Key::Down,
-                            Home => Key::Home,
-                            End => Key::End,
-                            PageUp => Key::PageUp,
-                            PageDown => Key::PageDown,
-                            Tab => Key::Char('\t'),
-                            BackTab => Key::BackTab,
-                            Delete => Key::Delete,
-                            Insert => Key::Insert,
-                            F(k) => Key::F(k),
-                            Null => Key::Null,
-                            Esc => Key::Esc,
-                            Char(c) => match key.modifiers {
-                                KeyModifiers::NONE | KeyModifiers::SHIFT => Key::Char(c),
-                                KeyModifiers::CONTROL => Key::Ctrl(c),
-                                KeyModifiers::ALT => Key::Alt(c),
-                                _ => Key::Null,
-                            },
-                        };
-                        tx.send(Event::Input(key)).unwrap();
+                    if event::poll(timeout).unwrap() {
+                        if let event::Event::Key(key) = event::read().unwrap() {
+                            let key = match key.code {
+                                Backspace => Key::Backspace,
+                                Enter => Key::Char('\n'),
+                                Left => Key::Left,
+                                Right => Key::Right,
+                                Up => Key::Up,
+                                Down => Key::Down,
+                                Home => Key::Home,
+                                End => Key::End,
+                                PageUp => Key::PageUp,
+                                PageDown => Key::PageDown,
+                                Tab => Key::Char('\t'),
+                                BackTab => Key::BackTab,
+                                Delete => Key::Delete,
+                                Insert => Key::Insert,
+                                F(k) => Key::F(k),
+                                Null => Key::Null,
+                                Esc => Key::Esc,
+                                Char(c) => match key.modifiers {
+                                    KeyModifiers::NONE | KeyModifiers::SHIFT => Key::Char(c),
+                                    KeyModifiers::CONTROL => Key::Ctrl(c),
+                                    KeyModifiers::ALT => Key::Alt(c),
+                                    _ => Key::Null,
+                                },
+                            };
+                            tx.send(Event::Input(key)).unwrap();
+                        }
+                    }
+
+                    if last_tick.elapsed() >= tick_rate && tx.send(Event::Tick).is_ok() {
+                        last_tick = Instant::now();
                     }
                 }
             })
         };
-        let _tick_handle = {
-            let tx = tx.clone();
-            let pause_ticker = pause_ticker.clone();
-            thread::spawn(move || loop {
-                // print!("\r\n");
-                // dbg!(*pause_ticker.lock().unwrap());
-                while pause_ticker.load(Ordering::SeqCst) {
-                    thread::sleep(Duration::from_millis(250));
-                }
-                if tx.send(Event::Tick).is_err() {
-                    break;
-                }
-                thread::sleep(config.tick_rate);
-            })
-        };
-
-        Events {
-            rx,
-            tx,
-            pause_stdin,
-            pause_ticker,
-        }
+        Events { rx, tx, pause_stdin }
     }
 
     /// Attempts to read an event.
     /// This function will block the current thread.
     pub fn next(&self) -> Result<Event<Key>, mpsc::RecvError> {
         self.rx.recv()
-    }
-
-    pub fn pause_ticker(&self) {
-        self.pause_ticker.swap(true, Ordering::SeqCst);
-    }
-
-    pub fn resume_ticker(&self) {
-        self.pause_ticker.swap(false, Ordering::SeqCst);
     }
 
     pub fn pause_event_loop(&self) {
@@ -162,14 +144,16 @@ impl Events {
 
     pub fn pause_key_capture(&self, terminal: &mut Terminal<CrosstermBackend<io::Stdout>>) {
         self.pause_event_loop();
+        thread::sleep(Duration::from_millis(250));
         disable_raw_mode().unwrap();
         execute!(io::stdout(), LeaveAlternateScreen, DisableMouseCapture).unwrap();
         terminal.show_cursor().unwrap();
     }
 
     pub fn resume_key_capture(&self, terminal: &mut Terminal<CrosstermBackend<io::Stdout>>) {
-        enable_raw_mode().unwrap();
         execute!(io::stdout(), EnterAlternateScreen, EnableMouseCapture).unwrap();
+        enable_raw_mode().unwrap();
+        thread::sleep(Duration::from_millis(250));
         self.resume_event_loop();
         terminal.resize(terminal.size().unwrap()).unwrap();
     }

